aspnetcore/fundamentals/websockets.md

```diff
@@ -14,17 +14,50 @@ uid: fundamentals/websockets
 
 This article explains how to get started with WebSockets in ASP.NET Core. [WebSocket](https://wikipedia.org/wiki/WebSocket) ([RFC 6455](https://tools.ietf.org/html/rfc6455)) is a protocol that enables two-way persistent communication channels over TCP connections. It's used in apps that benefit from fast, real-time communication, such as chat, dashboard, and game apps.
 
-[View or download sample code](https://github.com/dotnet/AspNetCore.Docs/tree/main/aspnetcore/fundamentals/websockets/samples) ([how to download](xref:index#how-to-download-a-sample), [how to run](#sample-app)).
+## Http/2 WebSockets support
+
+Using WebSockets over HTTP/2 takes advantage of new features such as:
+
+* Header compression.
+* Multiplexing, which reduces the time and resources needed when making multiple requests to the server.
+
+These supported features are available in Kestrel on all HTTP/2 enabled platforms. The version negotiation is automatic in browsers and Kestrel, so no new APIs are needed.
+
+.NET 7 introduced Websockets over HTTP/2 support for Kestrel, SignalR JavaScript client and SignalR with Blazor WebAssembly.
+
+> [!NOTE]
+> HTTP/2 WebSockets use CONNECT requests rather than GET, so your own routes and controllers may need updating.
+> Chrome and Edge have HTTP/2 WebSockets enabled by default, and you can enable it in FireFox on the about:config page with the network.http.spdy.websockets flag.
+
+WebSockets were originally designed for HTTP/1.1 but have since been adapted to work over HTTP/2. 
 
 ## SignalR
 
 [ASP.NET Core SignalR](xref:signalr/introduction) is a library that simplifies adding real-time web functionality to apps. It uses WebSockets whenever possible.
 
-For most applications, we recommend SignalR over raw WebSockets. SignalR provides transport fallback for environments where WebSockets isn't available. It also provides a basic remote procedure call app model. And in most scenarios, SignalR has no significant performance disadvantage compared to using raw WebSockets.
+For most applications, we recommend SignalR over raw WebSockets.
+
```

Comment:
The original text all in one paragraph made it clear that the "SignalR provides..." text explained the recommendation. Splitting it into separate paragraphs obscures that. I suggest restoring it as one paragraph consisting of the bullet points with an intro.

"over Websockets" could be interpreted as "using Websockets"

```suggestion
For most applications, we recommend SignalR rather than raw WebSockets.

```